### ISSUE TITLE
why is watsjs failing on PR#4950?

### ISSUE THREAD

**Issue by xtremerui** [concourse/concourse#5288]
We are seeing failures like:

```
 smoke › running pipelines

  /tmp/build/82e11db2/concourse/web/wats/node_modules/puppeteer/lib/DOMWorld.js:561

  Rejected promise returned by test. Reason:

  TimeoutError {
    message: 'waiting for selector "[data-step-name="hello"] .header" failed: timeout 30000ms exceeded',
  }

  new WaitTask (node_modules/puppeteer/lib/DOMWorld.js:561:28)
  DOMWorld._waitForSelectorOrXPath (node_modules/puppeteer/lib/DOMWorld.js:490:22)
  DOMWorld.waitForSelector (node_modules/puppeteer/lib/DOMWorld.js:444:17)
  Frame.waitForSelector (node_modules/puppeteer/lib/FrameManager.js:628:47)
  Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:112:23)
  Frame.waitFor (node_modules/puppeteer/lib/FrameManager.js:613:19)
  Page.waitFor (node_modules/puppeteer/lib/Page.js:1080:29)
  Web.clickAndWait (helpers/web.js:77:21)
  clickAndWait (test/smoke.js:36:23)



  build › shows abort hooks

  /tmp/build/82e11db2/concourse/web/wats/node_modules/puppeteer/lib/DOMWorld.js:561

  Rejected promise returned by test. Reason:

  TimeoutError {
    message: 'waiting for function failed: timeout 60000ms exceeded',
  }

  new WaitTask (node_modules/puppeteer/lib/DOMWorld.js:561:28)
  DOMWorld.waitForFunction (node_modules/puppeteer/lib/DOMWorld.js:466:12)
  Frame.waitForFunction (node_modules/puppeteer/lib/FrameManager.js:658:28)
  Frame.waitFor (node_modules/puppeteer/lib/FrameManager.js:618:19)
  Page.waitFor (node_modules/puppeteer/lib/Page.js:1080:29)
  waitFor (test/build.js:39:28)

 error Command failed with exit code 1.
 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

https://ci.concourse-ci.org/teams/main/pipelines/prs/jobs/watsjs/builds/676#L5e4cf268:312:313

**Comment by xtremerui** [comment]
We opened the Network tab of the Developer Tools in Chrome and copied the `Cookie:` header from the `api/v1/builds/<build id>/events` request. Then we did
```bash
$ curl -H "Cookie: <contents copied above> http://localhost:8080/api/v1/builds/1/events
curl: (52) Empty reply from server
```

And the web service was actually down! We checked the logs (`docker-compose logs web`), and saw:

```
web_1     | panic: interface conversion: *auth.responseWrapper is not http.Flusher: missing method Flush
web_1     |
web_1     | goroutine 1952 [running]:
web_1     | github.com/concourse/concourse/atc/api/buildserver.NewEventHandler.func1(0x401b100, 0xc0007d38e0, 0xc000022e00)
web_1     | 	/src/atc/api/buildserver/eventhandler.go:39 +0x5b7
web_1     | net/http.HandlerFunc.ServeHTTP(0xc000d2f8c0, 0x401b100, 0xc0007d38e0, 0xc000022e00)
web_1     | 	/usr/local/go/src/net/http/server.go:2007 +0x44
web_1     | github.com/concourse/concourse/atc/api/buildserver.(*Server).BuildEvents.func1.1(0xc0006caba0, 0xc0006383c0, 0x40902c0, 0xc000451680, 0x401b100, 0xc0007d38e0, 0xc000022e00)
web_1     | 	/src/atc/api/buildserver/events.go:16 +0x103
web_1     | created by github.com/concourse/concourse/atc/api/buildserver.(*Server).BuildEvents.func1
web_1     | 	/src/atc/api/buildserver/events.go:13 +0xbe
```

**Comment by pivotal-jamie-klassen** [comment]
Solved in 78a30c4
